### PR TITLE
tests for kubernetes program generation

### PR DESCRIPTION
**Pull request by EvanBoyle**
This creates a set of tests so that we work on fixing https://github.com/pulumi/pulumi/issues/5101

### CODE REVIEW

File: pkg/codegen/internal/test/testdata/kubernetes-pod.pp.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import pulumi
+import pulumi_kubernetes as kubernetes
+
+bar = kubernetes.core_v1.Pod("bar",
```
Comment (lblackstone) [inline pkg/codegen/internal/test/testdata/kubernetes-pod.pp.py:4]:
```suggestion
bar = kubernetes.core.v1.Pod("bar",
```

File: pkg/codegen/internal/test/testdata/kubernetes-pod.pp.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"github.com/pulumi/pulumi-kubernetes/sdk/v2/go/kubernetes/core/v1"
+	"github.com/pulumi/pulumi/sdk/v2/go/pulumi"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		_, err := core / v1.NewPod(ctx, "bar", &core/v1.PodArgs{
```
Comment (lblackstone) [inline pkg/codegen/internal/test/testdata/kubernetes-pod.pp.go:10]:
```suggestion
		_, err := corev1.NewPod(ctx, "bar", &core/v1.PodArgs{
```

File: pkg/codegen/internal/test/testdata/kubernetes-pod.pp.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"github.com/pulumi/pulumi-kubernetes/sdk/v2/go/kubernetes/core/v1"
```
Comment (lblackstone) [inline pkg/codegen/internal/test/testdata/kubernetes-pod.pp.go:4]:
```suggestion
	corev1 "github.com/pulumi/pulumi-kubernetes/sdk/v2/go/kubernetes/core/v1"
```

Comment (lblackstone) [inline pkg/codegen/internal/test/testdata/kubernetes-pod.pp.go:4]: Missing `metav1 "github.com/pulumi/pulumi-kubernetes/sdk/v2/go/kubernetes/meta/v1"`

File: pkg/codegen/internal/test/testdata/kubernetes-pod.pp.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	"github.com/pulumi/pulumi/sdk/v2/go/pulumi"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		_, err := core / v1.NewPod(ctx, "bar", &core/v1.PodArgs{
+			ApiVersion: pulumi.String("v1"),
+			Kind:       pulumi.String("Pod"),
+			Metadata: &meta.ObjectMetaArgs{
```
Comment (lblackstone) [inline pkg/codegen/internal/test/testdata/kubernetes-pod.pp.go:13]:
```suggestion
			Metadata: &metav1.ObjectMetaArgs{
```

File: pkg/codegen/internal/test/testdata/kubernetes-pod.pp.go [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+				Namespace: pulumi.String("foo"),
+				Name:      pulumi.String("bar"),
+			},
+			Spec: &core.PodSpecArgs{
```
Comment (lblackstone) [inline pkg/codegen/internal/test/testdata/kubernetes-pod.pp.go:17]:
```suggestion
			Spec: &corev1.PodSpecArgs{
```

File: pkg/codegen/internal/test/testdata/kubernetes-pod.pp.go [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+				Containers: core.ContainerArray{
```
Comment (lblackstone) [inline pkg/codegen/internal/test/testdata/kubernetes-pod.pp.go:18]:
```suggestion
				Containers: corev1.ContainerArray{
```

File: pkg/codegen/internal/test/testdata/kubernetes-pod.pp.go [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+					&core.ContainerArgs{
```
Comment (lblackstone) [inline pkg/codegen/internal/test/testdata/kubernetes-pod.pp.go:19]:
```suggestion
					&corev1.ContainerArgs{
```

File: pkg/codegen/internal/test/testdata/kubernetes-pod.pp.go [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+						Name:  pulumi.String("nginx"),
+						Image: pulumi.String("nginx:1.14-alpine"),
+						Resources: &core.ResourceRequirementsArgs{
```
Comment (lblackstone) [inline pkg/codegen/internal/test/testdata/kubernetes-pod.pp.go:22]:
```suggestion
						Resources: &corev1.ResourceRequirementsArgs{
```

**Comment by EvanBoyle** [comment]
Discussed with @lblackstone offline, these tests just capture current behavior so we can paralelize fixes.

**Pull request merged**